### PR TITLE
Users crud service

### PR DESCRIPTION
**Pull request by danilovIgor773**

### CODE REVIEW

File: src/controllers/users.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { Request, Response } from 'express';
+// import { v4 as uuid } from 'uuid';
```
Comment (FakhriddinAbdurakhimovEPAM) [inline src/controllers/users.ts]: please remove not needed code

File: src/controllers/users.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    if (!user) return res.status(404).send('The user with given id is not found');
+
+    console.log(`Retrieved the user with the following id: ${req.params.id}`);
+
+    res.send(user);
+};
+
+export const createUser = (req: Request, res: Response) => {
+    const user = req.body as User;
+    const { error } = validateUser(user);
```
Comment (FakhriddinAbdurakhimovEPAM) [inline src/controllers/users.ts]: what if two users have the same id, is there validation for this?

File: src/controllers/users.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    res.send(user);
+};
+
+export const deleteUser = (req: Request, res: Response) => {
+    const user = users.find((user) => user.id === req.params.id);
+
+    if (!user) return res.status(404).send('The user with given id is not found');
+
+    const deletedUser = deleteExistingUser(user);
```
Comment (FakhriddinAbdurakhimovEPAM) [inline src/controllers/users.ts:45]: isDeleted flag is true, but there is no filtration in getAllUsers, getUserById , updateUser, so this isDeleted: true does nothing, it's not soft delete

File: src/controllers/users.ts (hunk 2 above)
Comment (FakhriddinAbdurakhimovEPAM) [reply src/controllers/users.ts]: probably the same should be for "login"

**Review by FakhriddinAbdurakhimovEPAM — CHANGES_REQUESTED** [review]
Overall good, please complete all tasks
I could not find
```
get auto-suggest list from limitusers, sorted by login property and filtered by loginSubstringin the login property:
        getAutoSuggestUsers(loginSubstring, limit)
```

File: src/controllers/users.ts (hunk 1 above)
Comment (danilovIgor773) [reply src/controllers/users.ts]: resolved

File: src/controllers/users.ts (hunk 2 above)
Comment (danilovIgor773) [reply src/controllers/users.ts]: resolved

File: src/controllers/users.ts (hunk 3 above)
Comment (danilovIgor773) [reply src/controllers/users.ts:45]: resolved

**Pull request merged**